File: lib/screens/guest_home_screen.dart
import 'package:flutter/material.dart';
import '../components/quizora_components.dart';
import '../data/database_helper.dart';
import '../models/user_progress.dart';
import 'chapter_progress_screen.dart';
import 'profile_screen.dart';
import 'quiz_flow_selection_screen.dart';
import 'study_main_screen.dart';

enum HomeTab { home, notes, quiz, profile }

class GuestHomeScreen extends StatefulWidget {
  final VoidCallback onSwitchLanguageRequested;

  const GuestHomeScreen({Key? key, required this.onSwitchLanguageRequested}) : super(key: key);

  @override
  _GuestHomeScreenState createState() => _GuestHomeScreenState();
}

class _GuestHomeScreenState extends State<GuestHomeScreen> {
  HomeTab _activeTab = HomeTab.home;
  UserProgress _progress = UserProgress();
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _loadProgress();
  }

  Future<void> _loadProgress() async {
    final progress = await DatabaseHelper.instance.getProgress();
    setState(() {
      _progress = progress;
      _isLoading = false;
    });
  }

  @override
  Widget build(BuildContext context) {
    if (_isLoading) {
      return const Scaffold(
        backgroundColor: spaceBgStart,
        body: Center(child: CircularProgressIndicator(color: neonBlue)),
      );
    }

    return Scaffold(
      backgroundColor: Colors.transparent,
      body: GlowBackground(
        child: Column(
          children: [
            Expanded(
              child: AnimatedSwitcher(
                duration: const Duration(milliseconds: 220),
                child: _buildTabContent(),
              ),
            ),
          ],
        ),
      ),
      bottomNavigationBar: BottomNavigationBarWidget(
        activeTab: _activeTab,
        onTabSelected: (tab) {
          setState(() {
            _activeTab = tab;
          });
          _loadProgress();
        },
      ),
    );
  }

  Widget _buildTabContent() {
    switch (_activeTab) {
      case HomeTab.home:
        return HomeContent(
          progress: _progress,
          onSwitchLanguage: widget.onSwitchLanguageRequested,
          onTabSelected: (tab) {
            setState(() {
              _activeTab = tab;
            });
          },
          onNavigateToDailyChallenge: () {
            Navigator.push(
              context,
              MaterialPageRoute(
                builder: (_) => const StudyMainScreen(
                  subject: 'Science',
                  chapter: 'Daily Battle Mix',
                  isDailyChallenge: true,
                ),
              ),
            ).then((_) => _loadProgress());
          },
          onNavigateToChapterProgress: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (_) => const ChapterProgressScreen()),
            ).then((_) => _loadProgress());
          },
        );
      case HomeTab.notes:
        return const QuizFlowSelectionScreen(initialIsNotesFlow: true);
      case HomeTab.quiz:
        return const QuizFlowSelectionScreen(initialIsNotesFlow: false);
      case HomeTab.profile:
        return ProfileScreen(
          progress: _progress,
          onSwitchLanguage: widget.onSwitchLanguageRequested,
          onRefreshProgress: _loadProgress,
        );
    }
  }
}

class HomeContent extends StatelessWidget {
  final UserProgress progress;
  final VoidCallback onSwitchLanguage;
  final Function(HomeTab) onTabSelected;
  final VoidCallback onNavigateToDailyChallenge;
  final VoidCallback onNavigateToChapterProgress;

  const HomeContent({
    Key? key,
    required this.progress,
    required this.onSwitchLanguage,
    required this.onTabSelected,
    required this.onNavigateToDailyChallenge,
    required this.onNavigateToChapterProgress,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final List<String> badges = progress.badgesUnlocked
        .split(',')
        .map((e) => e.trim())
        .where((e) => e.isNotEmpty)
        .toList();

    return ListView(
      padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 24.0),
      children: [
        // --- 1. Header Segment ---
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  children: const [
                    Icon(Icons.electric_bolt, color: neonBlue, size: 24),
                    SizedBox(width: 6),
                    Text(
                      'QUIZORA',
                      style: TextStyle(
                        color: Colors.white,
                        fontSize: 24,
                        fontWeight: FontWeight.black,
                        letterSpacing: 2,
                      ),
                    ),
                  ],
                ),
                const Text(
                  'CBSE SMART LEARNING',
                  style: TextStyle(
                    color: neonPurple,
                    fontSize: 11,
                    fontWeight: FontWeight.bold,
                    letterSpacing: 1.0,
                  ),
                ),
              ],
            ),
            StatusShield(
              icon: Icons.translate,
              label: 'MEDIUM',
              value: progress.selectedLanguage.isEmpty ? 'English' : progress.selectedLanguage,
              glowColor: neonPurple,
              onClick: onSwitchLanguage,
            ),
          ],
        ),
        const SizedBox(height: 20),

        // --- 2. Xp and Streak cards Row ---
        Row(
          children: [
            Expanded(
              child: StatusShield(
                icon: Icons.star,
                label: 'STUDENT XP',
                value: '${progress.xp} XP',
                glowColor: neonBlue,
              ),
            ),
            const SizedBox(width: 10),
            Expanded(
              child: StatusShield(
                icon: Icons.local_fire_department,
                label: 'STREAK',
                value: '${progress.streak} Days',
                glowColor: Colors.orange,
              ),
            ),
          ],
        ),
        const SizedBox(height: 20),

        // --- 3. Prominent Daily challenge ---
        GlassCard(
          borderGlowColor: Colors.greenAccent,
          borderWidth: 1.5,
          child: Row(
            children: [
              Expanded(
                flex: 3,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: const [
                  Text(
                    'DAILY CHALLENGE',
                    style: TextStyle(
                      color: Colors.greenAccent,
                      fontSize: 16,
                      fontWeight: FontWeight.black,
                      letterSpacing: 0.5,
                    ),
                  ),
                  SizedBox(height: 4),
                  Text(
                    'Test your skills with a 10-Question mix for a bonus 50 XP!',
                    style: TextStyle(
                      color: textSecondary,
                      fontSize: 12,
                      height: 1.3,
                    ),
                  ),
                ],
              ),
              const SizedBox(width: 10),
              Expanded(
                flex: 2,
                child: NeonButton(
                  text: 'PLAY NOW',
                  onClick: onNavigateToDailyChallenge,
                  glowColor: Colors.greenAccent,
                ),
              ),
            ],
          ),
        ),
        const SizedBox(height: 20),

        // --- 4. Modules grid ---
        const Text(
          'LEARNING MODULES',
          style: TextStyle(
            color: textSecondary,
            fontSize: 14,
            fontWeight: FontWeight.bold,
            letterSpacing: 1.0,
          ),
        ),
        const SizedBox(height: 12),
        Column(
          children: [
            Row(
              children: [
                Expanded(
                  child: ModuleMenuCard(
                    title: 'Quiz / Tests',
                    description: 'CBSE Chapter MCQs with feedback and animations.',
                    icon: Icons.gamepad,
                    glowColor: neonBlue,
                    onClick: () => onTabSelected(HomeTab.quiz),
                  ),
                ),
                const SizedBox(width: 14),
                Expanded(
                  child: ModuleMenuCard(
                    title: 'Short Notes',
                    description: 'Definitions, formulas, textbook key highlights.',
                    icon: Icons.book,
                    glowColor: neonPurple,
                    onClick: () => onTabSelected(HomeTab.notes),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 14),
            Row(
              children: [
                Expanded(
                  child: ModuleMenuCard(
                    title: 'Daily Battle',
                    description: 'Randomized syllabus challenges and streak protectors.',
                    icon: Icons.bolt,
                    glowColor: Colors.orange,
                    onClick: onNavigateToDailyChallenge,
                  ),
                ),
                const SizedBox(width: 14),
                Expanded(
                  child: ModuleMenuCard(
                    title: 'My Progress',
                    description: 'Completed courses, revision trackers, metrics.',
                    icon: Icons.show_chart,
                    glowColor: Colors.greenAccent,
                    onClick: onNavigateToChapterProgress,
                  ),
                ),
              ],
            ),
          ],
        ),
        const SizedBox(height: 20),

        // --- 5. Active Badges ---
        GlassCard(
          borderGlowColor: textMuted,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text(
                'YOUR ACTIVE BADGES',
                style: TextStyle(
                  color: textSecondary,
                  fontSize: 12,
                  fontWeight: FontWeight.bold,
                  letterSpacing: 1.0,
                ),
              ),
              const SizedBox(height: 10),
              if (badges.isEmpty)
                const Text(
                  'Unlock badges by solving quizzes & scoring perfect points!',
                  style: TextStyle(color: textMuted, fontSize: 12),
                )
              else
                Wrap(
                  spacing: 8.0,
                  runSpacing: 8.0,
                  children: badges.map((badge) {
                    return Container(
                      padding: const EdgeInsets.symmetric(horizontal: 10.0, vertical: 6.0),
                      decoration: BoxDecoration(
                        color: neonPurple.withOpacity(0.15),
                        borderRadius: BorderRadius.circular(8.0),
                      ),
                      child: Row(
                        mainAxisSize: MainAxisSize.min,
                        children: [
                          const Icon(Icons.workspace_premium, color: Colors.orange, size: 14),
                          const SizedBox(width: 4),
                          Text(
                            badge,
                            style: const TextStyle(
                              color: Colors.white,
                              fontSize: 11,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                        ],
                      ),
                    );
                  }).toList(),
                ),
            ],
          ),
        ),
      ],
    );
  }
}

class ModuleMenuCard extends StatelessWidget {
  final String title;
  final String description;
  final IconData icon;
  final Color glowColor;
  final VoidCallback onClick;

  const ModuleMenuCard({
    Key? key,
    required this.title,
    required this.description,
    required this.icon,
    required this.glowColor,
    required this.onClick,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 140,
      decoration: BoxDecoration(
        color: glassBg,
        borderRadius: BorderRadius.circular(20.0),
        border: Border.all(
          color: glowColor.withOpacity(0.35),
          width: 1.0,
        ),
      ),
      child: Material(
        color: Colors.transparent,
        child: InkWell(
          borderRadius: BorderRadius.circular(20.0),
          onTap: onClick,
          child: Padding(
            padding: const EdgeInsets.all(16.0),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Icon(icon, color: glowColor, size: 28),
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      title,
                      style: const TextStyle(
                        color: Colors.white,
                        fontSize: 15,
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    const SizedBox(height: 2),
                    Text(
                      description,
                      style: const TextStyle(
                        color: textSecondary,
                        fontSize: 11,
                        height: 1.2,
                      ),
                      maxLines: 2,
                      overflow: TextOverflow.ellipsis,
                    ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

class BottomNavigationBarWidget extends StatelessWidget {
  final HomeTab activeTab;
  final Function(HomeTab) onTabSelected;

  const BottomNavigationBarWidget({
    Key? key,
    required this.activeTab,
    required this.onTabSelected,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Theme(
      data: Theme.of(context).copyWith(
        canvasColor: const Color(0xDC060410),
      ),
      child: BottomNavigationBar(
        currentIndex: _getTabIndex(activeTab),
        onTap: (index) {
          onTabSelected(_getTabFromIndex(index));
        },
        selectedItemColor: activeTab == HomeTab.notes || activeTab == HomeTab.profile ? neonPurple : neonBlue,
        unselectedItemColor: textSecondary,
        backgroundColor: const Color(0xDC060410),
        elevation: 8,
        type: BottomNavigationBarType.fixed,
        items: const [
          BottomNavigationBarItem(
            icon: Icon(Icons.home),
            label: 'Home',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.book),
            label: 'Notes',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.gamepad),
            label: 'Quiz',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.person),
            label: 'Profile',
          ),
        ],
      ),
    );
  }

  int _getTabIndex(HomeTab tab) {
    switch (tab) {
      case HomeTab.home:
        return 0;
      case HomeTab.notes:
        return 1;
      case HomeTab.quiz:
        return 2;
      case HomeTab.profile:
        return 3;
    }
  }

  HomeTab _getTabFromIndex(int index) {
    switch (index) {
      case 0:
        return HomeTab.home;
      case 1:
        return HomeTab.notes;
      case 2:
        return HomeTab.quiz;
      case 3:
        return HomeTab.profile;
      default:
        return HomeTab.home;
    }
  }
}

File: app/src/main/java/com/example/ui/screens/GuestHomeScreen.kt
package com.example.ui.screens

import androidx.compose.animation.*
import androidx.compose.animation.core.tween
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.*
import com.example.ui.theme.*
import com.example.viewmodel.QuizoraViewModel

enum class HomeTab {
    HOME, NOTES, QUIZ, PROFILE
}

@Composable
fun GuestHomeScreen(
    viewModel: QuizoraViewModel,
    onNavigateToDailyChallenge: () -> Unit,
    onNavigateToQuizFlow: () -> Unit,
    onNavigateToChapterProgress: () -> Unit,
    onNavigateToNotesSelector: () -> Unit,
    onSwitchLanguageRequested: () -> Unit
) {
    var activeTab by remember { mutableStateOf(HomeTab.HOME) }
    val progress by viewModel.userProgress.collectAsState()

    GlowBackground {
        Scaffold(
            containerColor = Color.Transparent,
            bottomBar = {
                BottomNavigationBar(
                    activeTab = activeTab,
                    onTabSelected = { activeTab = it }
                )
            },
            modifier = Modifier.fillMaxSize().windowInsetsPadding(WindowInsets.navigationBars)
        ) { paddingValues ->
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues)
            ) {
                // Animate content switches between tabs for smooth transitions
                AnimatedContent(
                    targetState = activeTab,
                    transitionSpec = {
                        fadeIn(animationSpec = tween(220)) togetherWith fadeOut(animationSpec = tween(220))
                    },
                    label = "tab_switches"
                ) { targetTab ->
                    when (targetTab) {
                        HomeTab.HOME -> HomeContent(
                            viewModel = viewModel,
                            onTabSelected = { activeTab = it },
                            onNavigateToDailyChallenge = onNavigateToDailyChallenge,
                            onNavigateToQuizFlow = onNavigateToQuizFlow,
                            onNavigateToChapterProgress = onNavigateToChapterProgress,
                            onNavigateToNotesSelector = onNavigateToNotesSelector,
                            onSwitchLanguage = onSwitchLanguageRequested
                        )
                        HomeTab.NOTES -> NotesScreenWrapper(
                            viewModel = viewModel,
                            onNavigateToStudy = { 
                                onNavigateToQuizFlow() 
                            }
                        )
                        HomeTab.QUIZ -> QuizFlowSelectionScreen(
                            viewModel = viewModel,
                            onNavigateToStudy = {
                                onNavigateToQuizFlow()
                            }
                        )
                        HomeTab.PROFILE -> ProfileScreen(
                            viewModel = viewModel,
                            onSwitchLanguage = onSwitchLanguageRequested
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun HomeContent(
    viewModel: QuizoraViewModel,
    onTabSelected: (HomeTab) -> Unit,
    onNavigateToDailyChallenge: () -> Unit,
    onNavigateToQuizFlow: () -> Unit,
    onNavigateToChapterProgress: () -> Unit,
    onNavigateToNotesSelector: () -> Unit,
    onSwitchLanguage: () -> Unit
) {
    val progress by viewModel.userProgress.collectAsState()

    LazyColumn(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 20.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp),
        contentPadding = PaddingValues(top = 24.dp, bottom = 24.dp)
    ) {
        // --- 1. Top Section ---
        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                        Icon(Icons.Default.ElectricBolt, contentDescription = "Quizora Flash", tint = NeonBlue)
                        Text(
                            text = "QUIZORA",
                            fontSize = 24.sp,
                            fontWeight = FontWeight.Black,
                            letterSpacing = 2.sp,
                            color = Color.White
                        )
                    }
                    Text(
                        text = "CBSE SMART LEARNING",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = NeonPurple,
                        letterSpacing = 1.sp
                    )
                }

                // Interactive language and class settings shortcuts
                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    StatusShield(
                        icon = Icons.Default.Translate,
                        label = "MEDIUM",
                        value = progress.selectedLanguage.ifEmpty { "English" },
                        glowColor = NeonPurple,
                        onClick = onSwitchLanguage
                    )
                }
            }
        }

        // --- 2. Gamification Metric Row ---
        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                Box(modifier = Modifier.weight(1f)) {
                    StatusShield(
                        icon = Icons.Default.Star,
                        label = "STUDENT XP",
                        value = "${progress.xp} XP",
                        glowColor = NeonBlue
                    )
                }
                Box(modifier = Modifier.weight(1f)) {
                    StatusShield(
                        icon = Icons.Default.LocalFireDepartment,
                        label = "STREAK",
                        value = "${progress.streak} Days",
                        glowColor = GoldStreak
                    )
                }
            }
        }

        // --- 3. Prominent Daily Challenge Alert ---
        item {
            GlassCard(
                borderGlowColor = NeonGreen,
                borderWidth = 1.5.dp
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(modifier = Modifier.weight(1.5f)) {
                        Text(
                            text = "DAILY CHALLENGE",
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Black,
                            color = NeonGreen,
                            letterSpacing = 0.5.sp
                        )
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            text = "Test your skills with a 10-Question mix for a bonus 50 XP!",
                            fontSize = 12.sp,
                            color = TextSecondary,
                            lineHeight = 16.sp
                        )
                    }
                    Spacer(modifier = Modifier.width(10.dp))
                    Box(modifier = Modifier.weight(1f)) {
                        NeonButton(
                            text = "PLAY NOW",
                            onClick = {
                                viewModel.startDailyChallenge()
                                onNavigateToDailyChallenge()
                            },
                            glowColor = NeonGreen,
                            testTag = "daily_challenge_button"
                        )
                    }
                }
            }
        }

        // --- 4. Main Activity Middle Grid ---
        item {
            Text(
                text = "LEARNING MODULES",
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = TextSecondary,
                letterSpacing = 1.sp,
                modifier = Modifier.padding(bottom = 8.dp)
            )
        }

        item {
            Column(verticalArrangement = Arrangement.spacedBy(14.dp)) {
                // Row 1
                Row(horizontalArrangement = Arrangement.spacedBy(14.dp)) {
                    Box(modifier = Modifier.weight(1f)) {
                        ModuleMenuCard(
                            title = "Quiz / Tests",
                            description = "CBSE Chapter MCQs with feedback and animations.",
                            icon = Icons.Default.Gamepad,
                            glowColor = NeonBlue,
                            onClick = {
                                onTabSelected(HomeTab.QUIZ)
                            }
                        )
                    }
                    Box(modifier = Modifier.weight(1f)) {
                        ModuleMenuCard(
                            title = "Short Notes",
                            description = "Definitions, formulas, textbook key highlights.",
                            icon = Icons.Default.Book,
                            glowColor = NeonPurple,
                            onClick = {
                                onTabSelected(HomeTab.NOTES)
                            }
                        )
                    }
                }

                // Row 2
                Row(horizontalArrangement = Arrangement.spacedBy(14.dp)) {
                    Box(modifier = Modifier.weight(1f)) {
                        ModuleMenuCard(
                            title = "Daily Battle",
                            description = "Randomized syllabus challenges and streak protectors.",
                            icon = Icons.Default.Bolt,
                            glowColor = GoldStreak,
                            onClick = {
                                viewModel.startDailyChallenge()
                                onNavigateToDailyChallenge()
                            }
                        )
                    }
                    Box(modifier = Modifier.weight(1f)) {
                        ModuleMenuCard(
                            title = "My Progress",
                            description = "Completed courses, revision trackers, metrics.",
                            icon = Icons.Default.ShowChart,
                            glowColor = NeonGreen,
                            onClick = onNavigateToChapterProgress
                        )
                    }
                }
            }
        }

        // --- 5. Custom Gamer Badges Teaser ---
        item {
            GlassCard(borderGlowColor = TextMuted) {
                Text(
                    text = "YOUR ACTIVE BADGES",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Bold,
                    color = TextSecondary,
                    letterSpacing = 1.sp
                )
                Spacer(modifier = Modifier.height(10.dp))
                
                val badges = progress.badgesUnlocked.split(",").map { it.trim() }.filter { it.isNotEmpty() }
                if (badges.isEmpty()) {
                    Text(
                        text = "Unlock badges by solving quizzes & scoring perfect points!",
                        fontSize = 12.sp,
                        color = TextMuted
                    )
                } else {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        badges.forEach { badge ->
                            Box(
                                modifier = Modifier
                                    .background(NeonPurple.copy(alpha = 0.15f), shape = RoundedCornerShape(8.dp))
                                    .padding(horizontal = 8.dp, vertical = 6.dp)
                            ) {
                                Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                                    Icon(Icons.Default.WorkspacePremium, contentDescription = badge, tint = GoldStreak, modifier = Modifier.size(14.dp))
                                    Text(text = badge, fontSize = 11.sp, fontWeight = FontWeight.Bold, color = Color.White)
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun ModuleMenuCard(
    title: String,
    description: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    glowColor: Color,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .height(140.dp)
            .clickable(onClick = onClick),
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = GlassBg),
        border = BorderStroke(1.dp, Brush.horizontalGradient(listOf(glowColor.copy(alpha = 0.5f), Color.Transparent)))
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            verticalArrangement = Arrangement.SpaceBetween
        ) {
            Icon(
                imageVector = icon,
                contentDescription = title,
                tint = glowColor,
                modifier = Modifier.size(28.dp)
            )
            Column {
                Text(
                    text = title,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Black,
                    color = Color.White
                )
                Spacer(modifier = Modifier.height(2.dp))
                Text(
                    text = description,
                    fontSize = 11.sp,
                    color = TextSecondary,
                    lineHeight = 14.sp,
                    maxLines = 2
                )
            }
        }
    }
}

@Composable
fun BottomNavigationBar(
    activeTab: HomeTab,
    onTabSelected: (HomeTab) -> Unit
) {
    NavigationBar(
        containerColor = Color(0xDC060410),
        tonalElevation = 8.dp,
        modifier = Modifier.navigationBarsPadding()
    ) {
        NavigationBarItem(
            icon = { Icon(Icons.Default.Home, contentDescription = "Home") },
            label = { Text("Home", fontSize = 11.sp) },
            selected = activeTab == HomeTab.HOME,
            onClick = { onTabSelected(HomeTab.HOME) },
            colors = NavigationBarItemDefaults.colors(
                selectedIconColor = NeonBlue,
                unselectedIconColor = TextSecondary,
                selectedTextColor = NeonBlue,
                unselectedTextColor = TextSecondary,
                indicatorColor = NeonBlue.copy(alpha = 0.15f)
            ),
            modifier = Modifier.testTag("nav_home")
        )
        NavigationBarItem(
            icon = { Icon(Icons.Default.Book, contentDescription = "Notes") },
            label = { Text("Notes", fontSize = 11.sp) },
            selected = activeTab == HomeTab.NOTES,
            onClick = { onTabSelected(HomeTab.NOTES) },
            colors = NavigationBarItemDefaults.colors(
                selectedIconColor = NeonPurple,
                unselectedIconColor = TextSecondary,
                selectedTextColor = NeonPurple,
                unselectedTextColor = TextSecondary,
                indicatorColor = NeonPurple.copy(alpha = 0.15f)
            ),
            modifier = Modifier.testTag("nav_notes")
        )
        NavigationBarItem(
            icon = { Icon(Icons.Default.Gamepad, contentDescription = "Quiz") },
            label = { Text("Quiz", fontSize = 11.sp) },
            selected = activeTab == HomeTab.QUIZ,
            onClick = { onTabSelected(HomeTab.QUIZ) },
            colors = NavigationBarItemDefaults.colors(
                selectedIconColor = NeonBlue,
                unselectedIconColor = TextSecondary,
                selectedTextColor = NeonBlue,
                unselectedTextColor = TextSecondary,
                indicatorColor = NeonBlue.copy(alpha = 0.15f)
            ),
            modifier = Modifier.testTag("nav_quiz")
        )
        NavigationBarItem(
            icon = { Icon(Icons.Default.Person, contentDescription = "Profile") },
            label = { Text("Profile", fontSize = 11.sp) },
            selected = activeTab == HomeTab.PROFILE,
            onClick = { onTabSelected(HomeTab.PROFILE) },
            colors = NavigationBarItemDefaults.colors(
                selectedIconColor = NeonPurple,
                unselectedIconColor = TextSecondary,
                selectedTextColor = NeonPurple,
                unselectedTextColor = TextSecondary,
                indicatorColor = NeonPurple.copy(alpha = 0.15f)
            ),
            modifier = Modifier.testTag("nav_profile")
        )
    }
}
